jax/api.py

```diff
@@ -762,11 +763,15 @@ def value_and_grad(fun: Callable, argnums: Union[int, Sequence[int]] = 0,
             "same shape as the arguments at positions {argnums}.")
 
   _check_callable(fun)
+  if np.isscalar(argnums):
```

Comment:
I'm curious if this would be better written as:
```
try:
  argnums = (operator.index(argnums),)
except TypeError:
  argnums = tuple(map(operator.index, argnums))
```

i.e., if `operator.index` is ultimately the judge of whether we consider something an index or not, why not just call it first rather than first looking at `np.isscalar`, which may or may not coincide?

Same everywhere.

Reply:
Why not just call `_ensure_index_tuple`?

Reply:
Because in this instance, the code further down branches on whether the input was a tuple or a scalar.

Reply:
But hang on, it doesn't do so in a meaningful way, as best I can tell? i.e., we could canonicalize it to a tuple and then remove the branch below? The only reason I can see not to do that might be to avoid a tuple allocation.